src/path/fs.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2018 The predicates-rs Project Developers.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fmt;
+use std::path;
+
+use path::{FileContent, FileTypePredicate};
+use Predicate;
+
+/// Predicate that compares file matches
+#[derive(Clone, Debug)]
+pub struct FileSystemEntryPredicate {
+    file_type: FileTypePredicate,
+    file_content: Option<FileContent>,
+}
+
+impl Predicate<path::Path> for FileSystemEntryPredicate {
+    fn eval(&self, path: &path::Path) -> bool {
+        if self.file_type.is_file() {
+            if !self.file_type.eval(path) {
+                false
+            } else {
+                self.file_content.clone().unwrap() == FileContent::new(path).unwrap()
+            }
+        } else {
+            self.file_type.eval(path)
+        }
+    }
+}
+
+impl fmt::Display for FileSystemEntryPredicate {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "var is {}", self.file_type)
```

Comment:
Should it be `file_type` or the path?  The downside being we'd have to store off the path